src/corelibs/U2View/src/ov_assembly/AssemblyModel.cpp

```diff
@@ -175,40 +169,47 @@ U2Region AssemblyModel::getGlobalRegion() {
 }
 
 qint64 AssemblyModel::getModelLength(U2OpStatus &os) {
-    if (NO_VAL == cachedModelLength) {
-        // try to set length from attributes
-        U2AttributeDbi *attributeDbi = dbiHandle.dbi->getAttributeDbi();
-        if (attributeDbi != NULL) {
-            U2IntegerAttribute attr = U2AttributeUtils::findIntegerAttribute(attributeDbi, assembly.id, U2BaseAttributeName::reference_length, os);
-            LOG_OP(os);
-            if (attr.hasValidId()) {
-                cachedModelLength = attr.value;
-            }
-            // ignore incorrect attribute value and remove corrupted attribute (auto-fix incorrectly converted ugenedb)
-            if (cachedModelLength == 0) {
+    if (cachedModelLength != NO_VAL) {
+        return cachedModelLength;
+    }
+    // Try to set 'cachedModelLength' from attributes.
+    QSet<U2DbiFeature> dbiFeatureSet = dbiHandle.dbi->getFeatures();
+    bool canReadAttributes = dbiFeatureSet.contains(U2DbiFeature_ReadAttributes);
+    bool canWriteAttributes = dbiFeatureSet.contains(U2DbiFeature_WriteAttributes);
```

Comment:
I can offer texting something in the log if we don't have write permission.

Reply:
This should be a valid case. For example, the assembly can be in the remote (public) database.